cloud/blockstore/vhost-server/request_aio.cpp

```diff
@@ -13,7 +14,36 @@ namespace {
 
 ////////////////////////////////////////////////////////////////////////////////
 
+template <bool (IEncryptor::*CryptoOperation)(
+    const TBlockDataRef& src,
+    const TBlockDataRef& dst,
+    ui64 blockIndex)>
+[[nodiscard]] bool DoCryptoOperation(
+    IEncryptor& encryptor,
+    TBlockDataRef src,
+    TBlockDataRef dst,
+    ui64 startSector)
+{
+    const size_t sectorCount = src.Size() / VHD_SECTOR_SIZE;
+
+    for (size_t i = 0; i < sectorCount; ++i) {
+        TBlockDataRef srcRef(src.Data() + i * VHD_SECTOR_SIZE, VHD_SECTOR_SIZE);
+        TBlockDataRef dstRef(dst.Data() + i * VHD_SECTOR_SIZE, VHD_SECTOR_SIZE);
+
+        if (IsAllZeroes(srcRef.Data(), srcRef.Size())) {
```

Comment:
Generally in crypto you don't want to do this kind of optimizations since, encrypting zeros to zeroes leaks information about the data and weakens the encryption.

Reply:
Вобшето мы же не даём гарантии что с нового диска всегда будут прочитаны нули, в первы раз перед записью. Помоюму в случае encrypted volume можно спокойно мусор возврашать. Или надо проверить что валидны блок никогда в нулевой не кодируется

Reply:
Давайте небольшую встречу и обсудим